### DISCUSSION TITLE
VR Example

### DISCUSSION THREAD

**Discussion by renhaoyeh** [enable3d/enable3d#194] [Help / Questions]
I use Oculus quest 2 to enter vr example
https://enable3d.io/examples/virtual-reality.html
when I look around, there have some broken
but when I recording it, the video seems pretty good
have anyone meet this before

**Comment by yandeu** [comment]
Do you experience the same issue if you play one of the official three.js VR examples?

For example: https://threejs.org/examples/?q=vr#webxr_vr_dragging

**Reply by renhaoyeh** [reply]
No, when I experience official three.js VR examples everything is fine
Because of my project recently need physics engine
So I using enable3D in my project, so the problem come out
It's kind like screen tearing

**Reply by yandeu** [reply]
Do you use the enable3d framework or just the physics extensions for three.js?

If you know three.js well, you could just use the physics extensions.

---

The WebXR code for the enable3d framework is just 100 lines. I guess it is a quick fix. https://github.com/enable3d/enable3d/blob/master/packages/threeGraphics/src/plugins/webxr.ts

Unfortunately I only have the Google Cardboard 🤣

**Reply by renhaoyeh** [reply]
I use whole enable3d framework, may I ask how to use enable3d like physics extensions, is there an example?
And another question is about the controller, is there anyway to change it to hand model?

**Reply by yandeu** [reply]
On the examples page you will find a section **As Physics Extension for three.js**.

**Reply by yandeu** [reply]
Did you make it work?

**Reply by renhaoyeh** [reply]
Yeah, it work pretty good, thanks for the solution
